hr_calendar_rest_time/models/resource_calendar.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.tools.float_utils import float_round
+
+
+class ResourceCalendar(models.Model):
+
+    _inherit = 'resource.calendar'
+
+    @api.onchange('attendance_ids')
+    def _onchange_hours_per_day(self):
+        if self.env.context.get('use_old_onchange_hours_per_day'):
+            return super()._onchange_hours_per_day()
+        attendances = self.attendance_ids.filtered(
```

Comment:
Why are you filtered attendance_ids not `date_from` and `date_to`?
I tested create `Starting Date` and `End Date`. 
It not used `Rest Time` to calculate `Average hour per day`

Reply:
Yes, I know this is an issue but it is the same way odoo does it in the core `resource` module.
At first I implemented the right solution taking into account the start and end date but it is not that easy. You would need a cron to check every day which lines are active are need to be used for the calculation and then update the value, since it is computed in an onchange.
Maybe we can add this behavior but it should probably be in another module.